### PR TITLE
um 4.1.0 (new formula)

### PR DESCRIPTION
**Pull request by sinclairtarget**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

I _am_ the author of this software, but I had not considered submitting it to homebrew-core until [this issue](https://github.com/sinclairtarget/um/issues/14) requesting submission was created.

This formula has been available for a while now via [my tap](https://github.com/sinclairtarget/homebrew-wst/blob/master/Formula/um.rb).

### CODE REVIEW

**Comment by fxcoudert** [comment]
```
  * Non-libraries were installed to "/usr/local/opt/um/lib"
    Installing non-libraries to "lib" is discouraged.
    The offending files are:
      /usr/local/opt/um/lib/um.rb
```

**Comment by commitay** [comment]
[`travis`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/travis.rb)

File: Formula/um.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  resource "kramdown" do
+    url "https://rubygems.org/gems/kramdown-1.17.0.gem"
+    sha256 "5862410a2c1692fde2fcc86d78d2265777c22bd101f11c76442f1698ab242cd8"
+  end
+
+  def install
+    ENV["GEM_HOME"] = libexec
+
+    resource("kramdown").stage do |context|
```
Comment (commitay) [inline Formula/um.rb]:
```ruby 
   resources.each do |r|
      r.verify_download_integrity(r.fetch)
      system "gem", "install", r.cached_download, "--ignore-dependencies",
             "--no-document", "--install-dir", libexec
   end
```

File: Formula/um.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    bin.install libexec/"bin/um"
+    bin.env_script_all_files(libexec/"bin", :GEM_HOME => ENV["GEM_HOME"])
+
+    bash_completion.install "um-completion.sh"
+    man1.install Dir["doc/man1/*"]
+  end
+
+  test do
+    shell_output("#{bin}/um version")
```
Comment (commitay) [inline Formula/um.rb]: `assert_match version.to_s, shell_output("#{bin}/um version")`

but a stronger test that verifies some functionally would be preferred.

File: Formula/um.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
+class Um < Formula
+  desc "Command-line utility for creating and maintaining personal man pages"
+  homepage "https://github.com/sinclairtarget/um"
+  url "https://github.com/sinclairtarget/um/archive/4.1.0.tar.gz"
+  sha256 "0606cd8da69618d508d06dee859dd1147a4d8846cdff57fb8958c71fe906523f"
+
+  bottle :unneeded
```
Comment (commitay) [inline Formula/um.rb]: This can be removed as we should bottle this.

File: Formula/um.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  depends_on "ruby"
```
Comment (commitay) [inline Formula/um.rb]: What version of Ruby does `um` need, can we use system Ruby?

**Comment by zmwangx** [comment]
Ping?

**Review by commitay — APPROVED** [review]

**Comment by commitay** [comment]
Thanks for the new formula @sinclairtarget!

**Pull request closed without merging**